src/nools/definition-preparation.js

```diff
@@ -4,10 +4,24 @@ Definition-preparation.js binds a value for `this` in all the functions within a
 This fascilitates simple data sharing between functions, and allows function logic to reference the definition itself.
 */
 
-function prepare(definition) {
+function prepare(definition, defaultResolvedIf, Utils) {
   var targetContext = {};
```

Comment:
I had tried using `let` here at first but got this error from eslint: 
```
  8:7  error  Parsing error: Unexpected token targetContext
```
This is probably because of the `src/nools/.eslintrc` which has:

```
"parserOptions": {
    "ecmaVersion": 5
  },
```
Should we be changing it?

Reply:
medic-conf can run modern javascript, but the code in nools runs client-side and needs tighter restrictions.

I forget why it is ecmaVersion 5 though and not `2017` like we use in modern config. We need to cater to lowest common denominator...

I'm not sure if these conversations are still relevant. May break compatibility for very very old android versions (?) or cht-core v2.x users (?) I doubt it matters, but it's a change worth signalling if you go that path.
https://github.com/medic/medic-conf/issues/109#issuecomment-433002346

Reply:
Right, we need to test the app in crosswalk for oldie Android. 
We can try `"ecmaVersion": 6` which introduces `let, cons and arrow functions`, then make a medic-android apk with xwalk build it and run it. But I think it can work fine.

Reply:
I suggest that upgrading the `ecmaVersion` be part of another issue and PR.

Reply:
👍